src/qibolab/platform/platform.py

```diff
@@ -321,11 +318,7 @@ def execute(
         # so that each acquisition command carries the info with it.
         integration_setup: IntegrationSetup = {}
         for qubit in self.qubits.values():
-            integration_setup[qubit.acquisition.name] = (
-                qubit.kernel,
-                qubit.threshold,
-                qubit.iq_angle,
-            )
+            integration_setup[qubit.acquisition.name] = qubit.kernel
```

Comment:
Can't we put the kernel as well together with `threshold` and `angle`?

It seems to me that they are all acquisition-related, so I don't see much of the point of having a separate entry. Am I mising something?

Reply:
Indeed, that was also my idea but in the end I forgot to implement (that's why I converted to draft). Only thing is that I remember discussing this (putting kernels in acquisition config) with @hay-k and he was against or had a different idea about where to put them, which is also why he didn't put them there in his PR. However, I don't remember what was his plan.

Reply:
`kernels` are moved to `AcquisitionConfig` in 6316c443feb9461241a019a30153f6235a24c662. I also moved the `FIXME` comment by @hay-k there and after reading, I believe the original plan was to pass kernels via the acquire pulse, once this is seperated from the probe pulse. I will leave this for #916.

Note that one side-effect of this move, that may affect serialization is: https://github.com/qiboteam/qibolab/blob/6316c443feb9461241a019a30153f6235a24c662/src/qibolab/serialize.py#L218-L225
is now more complicated than before, as I need to manually delete the kernels array from the dictionary, since it is not JSON serializable.

Reply:
> `kernels` are moved to `AcquisitionConfig` in [6316c44](https://github.com/qiboteam/qibolab/commit/6316c443feb9461241a019a30153f6235a24c662). I also moved the `FIXME` comment by @hay-k there and after reading, I believe the original plan was to pass kernels via the acquire pulse, once this is seperated from the probe pulse. I will leave this for #916.

Perfect. Much better to have in the acquisition config than on their own. It could make sense that they could belong to the pulses (provided hardware support for the feature - and not being fully suboptimal).

> is now more complicated than before, as I need to manually delete the kernels array from the dictionary, since it is not JSON serializable.

They are, I'm serializing them for the custom pulses.
https://github.com/qiboteam/qibolab/blob/a07f6ff9cbe1b901c802d9355df1141a4b255028/src/qibolab/serialize_.py#L12-L28
(though we could make it more optimal for storage, or decide a different scheme - I have been conservative here)

In any case, keep it as it is here, I'll take care of that in #971